climpred/prediction.py

```diff
@@ -26,7 +30,13 @@
 # --------------------------------------------#
 @is_xarray([0, 1])
 def compute_perfect_model(
-    ds, control, metric='pearson_r', comparison='m2e', add_attrs=True
+    ds,
+    control,
+    metric='pearson_r',
+    comparison='m2e',
+    dim=['member', 'init'],
```

Comment:
lists in kwargs can lead to unexpected behavior.

I think should do
`dim=None`

then in code
```
if dim is None:
    dim = ['member', 'init']
```

Reply:
but then in the short docstring (jupyer: func?) only dim=None is shown as the default.

Reply:
You can still clarify in the docstring that the default is ['member', 'init']. It's really bad practice to have lists/dictionaries as defaults in python as @ahuang11 said. It does some weird stuff. I think if you input your own list into the function it will modify the default keyword locally for awhile for instance.

Reply:
ok. changed.